Grafo.cpp

```diff
@@ -27,84 +30,233 @@ Grafo::~Grafo() {
             elementos += 2;
         }
     }
     else
-        cout<<"\n\tYa existe una ruta que conecta " << origen << " con " << destino;
-}
\ No newline at end of file
+        cout << "\n\tYa existe una ruta que conecta " << origen << " con " << destino;
+}
+
+Matriz<string>* Grafo::generarRecorrido()
+{
+    Matriz<string>* recorridoMatriz = new Matriz<string>(VACIO,elementos,elementos);//MATRIZ NxN CON TODOS NONE
+
+    for(int i = 0; i < elementos; i++)
+    {
+        for(int j = 0 ; j < elementos ; j++ )
+        {
+            if(i != j)
+            {
+                string aeropuerto = vertices->obtenerDato(j);
+                recorridoMatriz->modificarElemento(aeropuerto,i,j);
+            }
+        }
+
+    }
+
+    return recorridoMatriz;
+}
+
+Matriz<string>* Grafo::caminoMinimoPrecio(Matriz<int>* &matrizAux)
+{
+
+    precioMatriz->copiarMatriz(matrizAux);
+    Matriz<string>* recorridoMatriz = generarRecorrido();
+
+    for(int k = 0 ; k < elementos ; k ++) {
+        for(int i = 0 ;  i < elementos ; i ++) {
+            for(int j = 0 ; j < elementos ; j++) {
+
+                int distancia = precioMatriz->obtenerValor(i,k) + precioMatriz->obtenerValor(k,j);
+
+                if(i != j && distancia < precioMatriz->obtenerValor(i,j)) {
+                    precioMatriz->modificarElemento(distancia,i,j);
+                    string aeropuerto = vertices->obtenerDato(k);
+                    recorridoMatriz->modificarElemento(aeropuerto,i,j);
+                }
+            }
+        }
+    }
+    return recorridoMatriz;
+}
+
+
+Matriz<string>* Grafo::caminoMinimoTiempo(Matriz<float>* &matrizAux)
+{
+    tiempoMatriz->copiarMatriz(matrizAux);
+    Matriz<string>* recorridoMatriz = generarRecorrido();
+
+    for(int k = 0 ; k < elementos ; k ++) {
+        for(int i = 0 ;  i < elementos ; i ++) {
+            for(int j = 0 ; j < elementos ; j++) {
+
+                int distancia = precioMatriz->obtenerValor(i,k) + precioMatriz->obtenerValor(k,j);
+
+                if(i != j && distancia < precioMatriz->obtenerValor(i,j)) {
+                    precioMatriz->modificarElemento(distancia,i,j);
+                    string aeropuerto = vertices->obtenerDato(k);
+                    recorridoMatriz->modificarElemento(aeropuerto,i,j);
+                }
+            }
+        }
+    }
+
+    return recorridoMatriz;
+}
+
+
+void Grafo::mostrarEscalas(unsigned posOrigen, unsigned posDestino, Matriz<string>* &recorridoMatriz,Diccionario<string, Aeropuerto*>* &pDiccionario)
+{
+    string escala = recorridoMatriz->obtenerValor(posOrigen,posDestino);
+    while(escala != vertices->obtenerDato(posDestino))
+    {
+        cout << "\t--Escala en:";
+
+        Aeropuerto *pAeropuerto = pDiccionario->obtenerValor(escala);
+        cout << "\n\t"<<*pAeropuerto;
+
+        unsigned posEscala = vertices->obtenerPosicion(escala);
+        escala =  recorridoMatriz->obtenerValor(posEscala , posDestino);
+    }
```

Comment:
Habría que chequear antes de obtener el valor del diccionario que la clave exista (la precondición del método `obtenerValor()` es que la clave exista, sino existe explota todo.)
Si no existe el aeropuerto en el Diccionario podemos mostrar el código IATA.
```suggestion
    while(escala != vertices->obtenerDato(posDestino))
    {
        cout << "\t--Escala en:";
        
        if (aeropuertos->existe(escala))
            cout << *aeropuertos->obtenerValor(escala);
        else
            cout << "\n\t" << escala;

        unsigned posEscala = vertices->obtenerPosicion(escala);
        escala =  recorridoMatriz->obtenerValor(posEscala , posDestino);
    }
```